test/transcoding/FPGABufferLocation.ll

```diff
@@ -0,0 +1,54 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_fpga_buffer_location -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; RUN: llvm-spirv -spirv-text -r %t.spt -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: 2 Capability FPGABufferLocationINTEL
+; CHECK-SPIRV: 9 Extension "SPV_INTEL_fpga_buffer_location"
+; CHECK-SPIRV: 3 Name [[ARGA:[0-9]+]] "a"
+; CHECK-SPIRV: 3 Name [[ARGB:[0-9]+]] "b"
+; CHECK-SPIRV: 3 Name [[ARGC:[0-9]+]] "c"
+; CHECK-SPIRV: 4 Decorate [[ARGC]] BufferLocationINTEL 0
+; CHECK-SPIRV: 4 Decorate [[ARGA]] BufferLocationINTEL 1
+; CHECK-SPIRV: 4 Decorate [[ARGB]] BufferLocationINTEL 2
+
+; CHECK-SPIRV: 5 Function
+; CHECK-SPIRV: 3 FunctionParameter {{[0-9]+}} [[ARGA]]
+; CHECK-SPIRV: 3 FunctionParameter {{[0-9]+}} [[ARGB]]
+; CHECK-SPIRV: 3 FunctionParameter {{[0-9]+}} [[ARGC]]
+
+; CHECK-LLVM: define spir_kernel void @test{{.*}} !kernel_arg_buffer_location ![[BUFLOC_MD:[0-9]+]] {{.*}}
+; CHECK-LLVM: ![[BUFLOC_MD]] = !{i32 1, i32 2, i32 0}
+
+; ModuleID = 'buffer_location.cl'
+source_filename = "buffer_location.cl"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-unknown"
+
+; Function Attrs: norecurse nounwind readnone
+define spir_kernel void @test(i32 addrspace(1)* %a, float addrspace(1)* %b, i32 addrspace(1)* %c) local_unnamed_addr !kernel_arg_addr_space !3 !kernel_arg_access_qual !4 !kernel_arg_type !5 !kernel_arg_base_type !5 !kernel_arg_buffer_location !6 {
```

Comment:
It would be nice to have also non-pointer parameter to see how it is (not) processed.

Reply:
I will add this case.
IMO non-pointer parameters shall also be processed (as it's done in this implementation), I'll raise a spec issue.

Reply:
I got the answer, guess, I need to adjust the implementation.

Reply:
If this PR is not ready for merge convert it to draft.

Reply:
Done

Reply:
Expanded the test